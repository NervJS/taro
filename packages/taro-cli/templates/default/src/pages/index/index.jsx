<%if (['react', 'preact'].includes(framework)) {-%>
import { Component<% if (typescript) {%>, PropsWithChildren<%}%> } from 'react'
<%}-%>
import { View, Text } from '@tarojs/components'
import './index.<%= cssExt %>'

export default class <%= _.capitalize(pageName) %> extends <% if (typescript) {%>Component<PropsWithChildren><%} else {%>Component<%}%> {

  componentWillMount () { }

  componentDidMount () { }

  componentWillUnmount () { }

  componentDidShow () { }

  componentDidHide () { }

  render () {
    return (
      <View className='<%= pageName %>'>
        <Text>Hello world!</Text>
      </View>
    )
  }
}
