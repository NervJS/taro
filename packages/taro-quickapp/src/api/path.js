// copy from https://github.com/jinder/path/blob/master/path.js
export function resolve () {
  var resolvedPath = ''
  var resolvedAbsolute = false

  for (var i = arguments.length - 1; i >= -1 && !resolvedAbsolute; i--) {
    var path = (i >= 0) ? arguments[i] : '/'

    if (!path) {
      continue
    }

    resolvedPath = path + '/' + resolvedPath
    resolvedAbsolute = path[0] === '/'
  }

  // At this point the path should be resolved to a full absolute path, but
  // handle relative paths to be safe (might happen when process.cwd() fails)

  // Normalize the path
  resolvedPath = normalizeArray(resolvedPath.split('/'),
    !resolvedAbsolute).join('/')

  return ((resolvedAbsolute ? '/' : '') + resolvedPath) || '.'
}

function normalizeArray (parts, allowAboveRoot) {
  var res = []
  for (var i = 0; i < parts.length; i++) {
    var p = parts[i]

    // ignore empty parts
    if (!p || p === '.') { continue }

    if (p === '..') {
      if (res.length && res[res.length - 1] !== '..') {
        res.pop()
      } else if (allowAboveRoot) {
        res.push('..')
      }
    } else {
      res.push(p)
    }
  }

  return res
}
