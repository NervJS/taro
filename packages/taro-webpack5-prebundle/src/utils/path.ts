export const addLeadingSlash = (url = '') => (url.charAt(0) === '/' ? url : '/' + url)
export const addTrailingSlash = (url = '') => (url.charAt(url.length - 1) === '/' ? url : url + '/')
export function parsePublicPath (publicPath = '/') {
  return ['', 'auto'].includes(publicPath) ? publicPath :  addTrailingSlash(publicPath)
}