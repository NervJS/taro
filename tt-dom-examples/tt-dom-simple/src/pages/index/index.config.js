// eslint-disable-next-line no-undef
export default definePageConfig({
  navigationBarTitleText: '首页'
})
