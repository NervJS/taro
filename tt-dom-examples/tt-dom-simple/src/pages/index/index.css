.grid-container {
    display: flex;
    flex-wrap: wrap;
    width: 100vw;
  }
  
.grid-box {
    width: 60px;
    height: 60px;
    margin: 3px;
    padding: 0;
}

.font {
    font-size: 24px;
}

.controls {
    text-align: center;
}


.button {
    display: inline-block;
    width: 150px;
    padding: 10px 20px;
    font-size: 16px;
    cursor: pointer;
    border: none;
    border-radius: 5px;
    margin-left: 20px;
    margin-bottom: 10px;
    color: white;
    transition: background-color 0.3s ease;
}



.start {
    background-color: #28a745; /* 绿色 */
}

.start:hover {
    background-color: #218838;
}

.clear {
    background-color: #dc3545; /* 红色 */
}

.clear:hover {
    background-color: #c82333;
}