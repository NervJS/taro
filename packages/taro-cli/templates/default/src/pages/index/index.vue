<template>
  <view class="<%= pageName %>">
    <text>{{ msg }}</text>
  </view>
</template>

<script>
<%if (framework === 'vue3') {-%>
import { ref } from 'vue'
<%}-%>
import './index.<%= cssExt %>'

export default {
<%if (framework === 'vue') {-%>
  data () {
    return {
      msg: 'Hello world!'
    }
  }
<%}-%>
<%if (framework === 'vue3') {-%>
  setup () {
    const msg = ref('Hello world')
    return {
      msg
    }
  }
<%}-%>
}
</script>
