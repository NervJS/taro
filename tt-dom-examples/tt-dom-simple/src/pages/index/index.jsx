import './index.css'

import { View } from '@tarojs/components'
import { useEffect, useRef, useState } from 'react'

const TOTAL = 1000
let timer = -1

const GridPage = () => {
  const [colors, setColors] = useState(
    () => new Array(TOTAL).fill(false) // false = 灰色，true = 蓝色
  )
  const currentIndex = useRef(0)

  useEffect(() => {
    console.log(' useEffect ')
  }, [])

  function start() {
    if (timer > 0) return
    timer = setInterval(() => {
      if (currentIndex.current >= TOTAL) {
        clearInterval(timer)
        timer = -1
        currentIndex.current = 0
        return
      }

      setColors((prev) => {
        const updated = [...prev]
        updated[currentIndex.current] = true
        currentIndex.current += 1
        return updated
      })
    }, 20)
  }

  function clear() {
    setColors(() => new Array(TOTAL).fill(false))
  }
  return (
    <View>
      <View className="controls">
        <View className="button start" onClick={start}>开始</View>
        <View className="button clear" onClick={clear}>清除</View>
      </View>
      <View className="grid-container">
        {colors
          .map((isActive, index) => (
            <View
              key={index}
              className="grid-box"
            >
              <View>
                <View>
                  <View>
                    <View style={{ backgroundColor: isActive ? '#3498db' : '#eee' }}>
                      <View className="font">{index}</View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          ))
        }
      </View>
    </View>

  )
}

export default GridPage
