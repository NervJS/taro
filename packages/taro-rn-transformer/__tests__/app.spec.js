import * as path from 'path'
import * as transformer from '../dist/index'

const App = `class HomeScreen extends React.Component {
  render () {
    return this.porps.children
  }
}`

function run () {
  const options = {
    projectRoot: path.resolve(__dirname),
    isEntryFile: () => { return true },
    nextTransformer: ({ src }) => { return src }
  }
  const code = transformer.transform({ src: App, filename: './src/app', options })
  return code
}

describe('app-loader', () => {
  it('app', () => {
    const result = run()
    expect(result).toMatchSnapshot()
  })
})
