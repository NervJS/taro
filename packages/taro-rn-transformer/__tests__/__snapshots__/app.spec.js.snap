// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-loader app 1`] = `
"import 'react-native/Libraries/polyfills/error-guard'
  import { AppRegistry } from 'react-native'
  import { createReactNativeApp } from '@tarojs/runtime-rn'
  import { createPageConfig } from '@tarojs/runtime-rn'
  import Component from './src/app'
  import pagesIndexIndex from './src/pages/index/index';import pagesExpoIndex from './src/pages/expo/index';import pagesAppIndex from './src/pages/app/index';import pagesAppText from './src/pages/app/text';import package1PagesSubTest1 from './src/package1/pages/sub/test1';import package1PagesSubApp from './src/package1/pages/sub/app'
  

  var config = {\\"appConfig\\":{\\"pages\\":[\\"pages/index/index\\",\\"pages/expo/index\\",\\"pages/app/index\\",\\"pages/app/text\\",\\"package1/pages/sub/test1\\",\\"package1/pages/sub/app\\"],\\"subPackages\\":[{\\"root\\":\\"package1\\",\\"pages\\":[\\"pages/sub/test1\\",\\"pages/sub/app\\"]}],\\"tabBar\\":{\\"backgroundColor\\":\\"#ffffff\\",\\"color\\":\\"#999\\",\\"selectedColor\\":\\"#ff552e\\",\\"list\\":[{\\"pagePath\\":\\"pages/index/index\\",\\"iconPath\\":\\"image/icon_component.png\\",\\"selectedIconPath\\":\\"image/icon_component_HL.png\\",\\"text\\":\\"首页\\"},{\\"pagePath\\":\\"pages/expo/index\\",\\"iconPath\\":\\"image/icon_API.png\\",\\"selectedIconPath\\":\\"image/icon_API_HL.png\\",\\"text\\":\\"expo\\"}]},\\"designWidth\\":750,\\"deviceRatio\\":{\\"640\\":1.17,\\"750\\":1,\\"828\\":0.905}}}
  global.__taroAppConfig = config
  config['pageList'] = [{name:'pagesIndexIndex',pagePath:'/pages/index/index',component:createPageConfig(pagesIndexIndex,{\\"pagePath\\":\\"/pages/index/index\\"})},{name:'pagesExpoIndex',pagePath:'/pages/expo/index',component:createPageConfig(pagesExpoIndex,{\\"pagePath\\":\\"/pages/expo/index\\"})},{name:'pagesAppIndex',pagePath:'/pages/app/index',component:createPageConfig(pagesAppIndex,{\\"pagePath\\":\\"/pages/app/index\\"})},{name:'pagesAppText',pagePath:'/pages/app/text',component:createPageConfig(pagesAppText,{\\"pagePath\\":\\"/pages/app/text\\"})},{name:'package1PagesSubTest1',pagePath:'/package1/pages/sub/test1',component:createPageConfig(package1PagesSubTest1,{\\"pagePath\\":\\"/package1/pages/sub/test1\\"})},{name:'package1PagesSubApp',pagePath:'/package1/pages/sub/app',component:createPageConfig(package1PagesSubApp,{\\"pagePath\\":\\"/package1/pages/sub/app\\"})}]
  AppRegistry.registerComponent('taroDemo',() => createReactNativeApp(Component,config))
  "
`;
